model/lifecycle.go

```diff
@@ -1633,7 +1635,8 @@ func addNewBuilds(ctx context.Context, creationInfo TaskCreationInfo, existingBu
 			},
 		)
 	}
-	if err = task.UpdateSchedulingLimit(creationInfo.Version.Author, creationInfo.Version.Requester, len(newActivatedTaskIds), true); err != nil {
+	numTasksModified := numEstimatedActivatedGeneratedTasks + len(newActivatedTaskIds)
```

Comment:
Should there be a unit test that this correctly estimates the number of tasks activated by generate.tasks?

Reply:
[This unit test](https://github.com/evergreen-ci/evergreen/blob/main/model/task/generate_tasks_estimations_test.go#L16) exists to test the accuracy of the `EstimatedNumActivatedGeneratedTasks` field. Added a new unit test asserting that this function counts the activated tasks correctly under the assumption that that field is accurate.